Stacks 2.1 updates for Clarinet, Stacks JS and API docs (#220)
hirosystems/docs: docs/stacks.js/how-to-guides/how-to-migrate-stacking-to-2.1.md

- [`stack-extend`](https://github.com/stacksgov/sips/blob/4ed252c11ade5569ab11a28d373b267f0e5499d4/sips/sip-015/sip-015-network-upgrade.md#new-method-stack-extend) = `client.stackExtend()`
- [`stack-increase`](https://github.com/stacksgov/sips/blob/4ed252c11ade5569ab11a28d373b267f0e5499d4/sips/sip-015/sip-015-network-upgrade.md#new-method-stack-increase) = `client.stackIncrease()`
- [`delegate-stack-extend`](https://github.com/stacksgov/sips/blob/4ed252c11ade5569ab11a28d373b267f0e5499d4/sips/sip-015/sip-015-network-upgrade.md#new-method-delegate-stack-extend) = `client.delegateStackExtend()`
- [`delegate-stack-increase`](https://github.com/stacksgov/sips/blob/4ed252c11ade5569ab11a28d373b267f0e5499d4/sips/sip-015/sip-015-network-upgrade.md#new-method-delegate-stack-increase) = `client.delegateStackIncrease()`

### New Helper Methods

<!-- todo: add links to api reference, once live -->

- `StackingClient.getAccountBalanceLocked()` gets the locked balance of the address from a node (`/v2` endpoint)
- `StackingClient.getAccountExtendedBalance()` gets more detailed balances of the address from an API (`/extended` endpoint); this includes STX, FTs, and NFTs

> - `StackingClient.getDelegationStatus()` gets the `get-delegation-info` information of the address from a node (readonly contract call)

Reviewer: ```suggestion
- `StackingClient.getDelegationStatus()` gets the `get-delegation-info` information of the address from a node (read-only contract call)
```

- `StackingClient.getPoxOperationInfo()` gets information about the current period of transition to PoX-2

## Migration

<!-- todo: version number is not fixated yet, might change -->

> Previous `@stacks/stacking` releases should automatically switch to the new PoX contract (in Period 3).

Reviewer: ```suggestion
Previous `@stacks/stacking` releases should automatically switch to the new PoX contract (in Period 3); however, if you want to use the new methods detailed above, you need to update to `@stacks/stacking` to version >=`6.0.0`.
```

> However, if you want to use the new methods detailed above, you need to update to `@stacks/stacking` to a version >=`6.0.0`.

Reviewer: ```suggestion
```
